### ISSUE TITLE
Attempting to add a package returns to package listing with no error or added package

### ISSUE THREAD

**Issue by Delinson** [SolderIO/solder#61]
I set a name of the package, and a slug then hit add a package. It will then take me back to the library page, but with nothing added, even checked the database and nothing has gone there either. I have made sure the permissions were set correctly for the public folder, along with config and app. Nothing just seems to be happening, am I missing something? Or is this an issue? (could not find anyone else having this issue)

**Comment by Indemnity83** [comment]
What was the name and slug you’re trying to use? It’s possible that there is a validation error missing or being triggered but not displayed.

**Comment by Delinson** [comment]
I tried DankNull which put in the slug of danknull, I checked all the error logs, there is no errors at all. Everything else writes to the database, I am completely lost as what else to try. Everything but adding mods works.

**Comment by Delinson** [comment]
Update: I did a complete reinstall, and still have the same issue. No idea why still. No errors in any logs, and no error posts when it happens.

**Comment by 5t0rmr1d3r** [comment]
i have the same issue with not adding a new package.
at the moment i create a database entry manual.

**Comment by Indemnity83** [comment]
Can one of you try this for me:

1. Make sure that debugging is on in your `.env` file:
```
APP_DEBUG=true
APP_ENV=local
```
2. Clear out any old log data by removing `storage/logs/laravel.log`
3. Attempt to create a new package. 
4. I expect this to have generated an error and recorded it in `storage/logs/laravel.log`, please post the contents of that file. 

Thanks

**Comment by Indemnity83** [comment]
I've added a readme to clarify the install process; I'm suspecting that the missing piece was linking the storage directory with the public folder with `php artisan storage:link`

**Comment by 5t0rmr1d3r** [comment]
i set in the env file debug=true
and tried to create a new package, but nothing happens. Also no log file is written.

**Comment by Indemnity83** [comment]
Can you confirm that you've linked the storage using the artisan command `php artisan storage:link`?

**Comment by 5t0rmr1d3r** [comment]
jep storage is linked via php artisan storage:link
![grafik](https://user-images.githubusercontent.com/4589916/35590327-6c24fafe-05fe-11e8-9987-ec8fb8eae312.png)

**Comment by Indemnity83** [comment]
Ok, stepping through the code; are you entering a website url, donation url or description too?

**Comment by 5t0rmr1d3r** [comment]
i'm entering everything except the donation url...

**Comment by Indemnity83** [comment]
What happens if you only do the name and slug? There was a bug in the other fields where you could potentially submit too long of a string and it may not have been caught in validation.

**Comment by Delinson** [comment]
I currently went back to 0.7.3... but when I had done this, I did just the name and slug and got nothing. But since I do not a current install atm (working on putting one on my coding environment) I could not give you if there was error logs.

**Comment by 5t0rmr1d3r** [comment]
nope also nothing, and no log.

**Comment by Indemnity83** [comment]
Well, I've tracked down one possible bug

Did you by chance use solder/upgrade package to migrate to v0.8? Looks like there is a bug in it where some of the database fields are not being set nullable properly. See solderio/upgrade#1

If you're able to; open up your database tables and make the author, description and website_url columns nullable and give it a try.

**Comment by Delinson** [comment]
Mine was a fresh install when it wasn't working. But I will let you know more tomorrow when I do a fresh install on my development environment.

**Comment by Indemnity83** [comment]
@5t0rmr1d3r can you check your db to see if the fields listed above are not set to nullable?

**Comment by 5t0rmr1d3r** [comment]
@Indemnity83 the fields are set to nulable...
![grafik](https://user-images.githubusercontent.com/4589916/35698410-d9e969a8-0784-11e8-87a5-d872120ae0be.png)

**Comment by Indemnity83** [comment]
I'm out of ideas ... Is there any possibility you can email me with a login and I can try to diagnose the system directly?

**Comment by cfarence** [comment]
@Indemnity83 I have the same issue, fresh install as well. Checked the database and those columns listed above are nullable also the storage directory has been linked. I've also set the debug to true and removed the log file and attempted to create a package again but no log file is written.

I've watched my logs in chrome and I can see the post request get sent to /library/. Now I'm not sure what the expected behavior here is but instead of returning status code 201 created it returns 301 Perm Redirect. I haven't yet dived into the code yet to look but I thought it was worth mentioning.

I'm running this on Apache 2.4.18 on Ubuntu 16.04

**Comment by Indemnity83** [comment]
Is it possibly a permissions error? Laravel 5 requires `/storage` folder to be writable. While not the proper permissions for a production system you could change the `/storage` folder to globally writable and see if that resolves the issue?

```
chmod 777 storage -R
```

**Comment by cfarence** [comment]
I've confirmed that the storage folder is writable by apache. The folder is owned by the same user and group that apache is running as. Though I did set storage directory to 777 and still the same issue. No log files are created and no files or database entries are created either.

I've had to move back to the previous version so some users could download the mod pack hosted on my instance, but I will be setting up a test instance of the beta version and see what happens.
